### PR TITLE
TDEVOPS-539 | double commit check

### PR DESCRIPTION
**Pull request by varenyatessell**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

Dependent Services #
- [ ] Governance
- [ ] IAM
- [ ] Deployer
- [ ] DB Systems
- [ ] Tenant
- [ ] Software Library
- [ ] Backup
- [ ] Conductor
- [ ] Monitoring
- [ ] DMM

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Environment**:

- [ ] Dev
- [ ] Staging

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/dev-checks.yml [hunk 1]
```diff
@@ -90,7 +94,14 @@ jobs:
            echo "Branch name does not start with 'TS-' or 'revert-' or 'TDEVOPS-'"
            exit 1
          fi
-     
+      - name: Check double commit
+        run: |
+          if ["${{ github.event.pull_request.user.login }}" != "cipipelinetessell" ] && [[ "${{ github.event.pull_request.head.ref }}" == *"double_commit"* ]] || [[ "${{ github.event.pull_request.title }}" == *"Double Commit"* ]]; then
```
Comment (sundeeptessell) [inline .github/workflows/dev-checks.yml:99]: Use {{ secret. CIPIPELINE_USER_NAME }}

File: .github/workflows/dev-checks.yml [hunk 2]
```diff
@@ -23,6 +23,10 @@ on:
         required: false
         type: string
         default: 'true'
+      check_double_commit:
```
Comment (sundeeptessell) [inline .github/workflows/dev-checks.yml:26]: Where is this conditional variable used?

File: .github/workflows/dev-checks.yml [hunk 3]
```diff
@@ -90,7 +94,14 @@ jobs:
            echo "Branch name does not start with 'TS-' or 'revert-' or 'TDEVOPS-'"
            exit 1
          fi
-     
+      - name: Check double commit
```
Comment (sundeeptessell) [inline .github/workflows/dev-checks.yml:97]: If this check is supposed to be conditional based on value of `check_double_commit` input, then, you need to split it into two. 
The login username check is MUST and always. It is not user controlled. The second part, which is to match the double commit string in title, etc.

File: .github/workflows/dev-checks.yml [hunk 4]
```diff
@@ -90,7 +94,14 @@ jobs:
...
+        run: |
+          if ["${{ github.event.pull_request.user.login }}" != "cipipelinetessell" ] && [[ "${{ github.event.pull_request.head.ref }}" == *"double_commit"* ]] || [[ "${{ github.event.pull_request.title }}" == *"Double Commit"* ]]; then
+              echo "Exclude Double commit in Branch, PR title and try again."
+              exit 1
+          else
+            echo "Clear"
```
Comment (sundeeptessell) [inline .github/workflows/dev-checks.yml:103]: Use better message.

**Review by sundeeptessell — CHANGES_REQUESTED** [review]

File: .github/workflows/dev-checks.yml (hunk 3 above)
Comment (varenyatessell) [reply .github/workflows/dev-checks.yml:97]: Looking into it there's no need for a conditional and this check must be followed in every case, so I'll just clear up the conditional

**Review by sundeeptessell — DISMISSED** [review]

**Review by sundeeptessell — APPROVED** [review]

**Pull request merged**